src/pages/index.js

```diff
@@ -103,70 +116,176 @@ const settings = {
+  (avatarData) => {
+    avatarModalPopup.setButtonText(true);
+    api
+      .editUserPhoto(avatarData)
+      .then((data) => {
+        userInfo.setAvatar(data.avatar);
+        avatarModalPopup.close();
+      })
+      .catch((err) => {
+        console.error(err);
+      })
+      .finally(() => avatarModalPopup.setButtonText());
+  }
+);
+
+const avatarModalElement = avatarModal.querySelector(".modal__form");
+
+/*----------------------------------*/
+/*      card like/remove likes      */
+/*----------------------------------*/
+
+// const handleLikeCard = (card) => {
+//   if (card._isliked) {
+//     api
+//       .removeCardLike(card._cardId)
+//       .then(() => {
+//         card.setLikes(false);
+//       })
+//       .catch((err) => {
+//         console.error(err);
+//       });
+//   } else {
+//     api
+//       .addCardLike(card._cardId)
+//       .then(() => {
+//         card.setLikes(true);
+//       })
+//       .catch((err) => {
+//         console.error(err);
+//       });
+//   }
+// };
+
 /*----------------------*/
 /*      popup items     */
 /*----------------------*/
 
-const editFormValidator = new FormValidator(settings, editFormElement);
-const addFormValidator = new FormValidator(settings, addFormElement);
-
-editFormValidator.enableValidation();
-addFormValidator.enableValidation();
-
-const addCardPopup = new PopupWithForm(
-  "#add-card-modal",
-  handleAddCardFormSubmit
-);
+const addCardPopup = new PopupWithForm("#add-card-modal", (cardData) => {
+  addCardPopup.setButtonText(true);
+  return api
+    .addCard(cardData)
+    .then((res) => {
+      console.log(res);
+      createCard(res);
+    })
+    .catch((error) => {
+      console.error(error);
+    });
+});
 addCardPopup.setEventListeners();
 
-const editProfilePopup = new PopupWithForm(
-  "#edit-modal",
-  handleProfileFormSubmit
-);
+const editProfilePopup = new PopupWithForm("#edit-modal", (card) => {
+  editProfilePopup.setButtonText(true);
+  return api
+    .editUserInfo(card)
+    .then((res) => {
+      console.log(res);
+      userInfo.setUserInfo(res.name, res.about);
+    })
+    .catch((error) => {
+      console.error(error);
+    });
+});
 editProfilePopup.setEventListeners();
 
-//function renderCard(cardData, wrapper) {
-//const newCard = new Card(cardData, cardSelector, handleCardClick);
-//cardSection.addItem(newCard.getView());
-//}
+avatarModalPopup.setEventListeners();
 
-const cardSection = new Section(
-  {
-    items: initialCards,
-    renderer: createCard,
-  },
-  cardListSelector
-);
+const editFormValidator = new FormValidator(settings, editFormElement);
+const addFormValidator = new FormValidator(settings, addFormElement);
 
-cardSection.renderItems();
+const editAvatarValidator = new FormValidator(settings, avatarModalElement);
+
+editFormValidator.enableValidation();
+addFormValidator.enableValidation();
+editAvatarValidator.enableValidation();
+
+Promise.all([api.getUserInfo(), api.getInitialCards()])
```

Comment:
Good job that you get all info with `Promise.all`

Reply:
thank you!